eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -712,6 +741,209 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
           goto(NORMAL) using d.copy(channelUpdate = channelUpdate1) storing()
       }
 
+    case Event(cmd: CMD_SPLICE, d: DATA_NORMAL) =>
+      val replyTo = if (cmd.replyTo == ActorRef.noSender) sender() else cmd.replyTo
+      d.spliceStatus match {
+        case SpliceStatus.NoSplice =>
+          if (d.commitments.isIdle && d.commitments.params.channelFeatures.hasFeature(DualFunding)) {
```

Comment:
> We should introduce a SplicePrototype feature, because we'll want to activate dual funding before splicing (and we'll also want to do splicing on top of single-funding channels). It's really important that we make sure we don't let peers trigger our splice code before we're ready to activate it!

Agreed, will add later.

> We should probably also always be the one initiating splices with Phoenix (because in the future that will let us aggregate operations on top of user-initiated splice-in using RBF), so for extra safety we should disable the handler for SpliceInit except when we're on regtest, this way we're sure that we can only use splice when we initiate it.

Good idea, done in ff0eb797e.

Reply:
Perfect, simply filtering to accept `splice_init` on regtest is actually good enough, no need to introduce a feature bit yet.